articles/article-005-array-at-zero-or-one.md

```diff
@@ -0,0 +1,36 @@
+<!-- BEGIN TITLE -->
+# Should array indices start at 0 or 1?
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+
+
+> “Should array indices start at 0 or 1? 
+> My compromise of 0.5 was rejected without, I thought, proper consideration.” 
+> -Stan Kelly-Bootle
+
+An array is a data structure to store some elements with an index for each. 
+This coins the question: 'what is the smallest index?'
+
+Most programming languages start an array at index 0 because it is the smallest natural number that can be represented in a binary system. But there are some rare outliers that start with index 1. 
+
+If you want to figure out whether the person in front of you is a computer scientist or not, there is one easy way to find it out: 
+Show a bunch of things an let they count them. The true computer scientist will start to count with a zero.
```

Comment:
Most people start counting with 1 (unless you live in a weird world where having one thing means having zero things?). I think you mean to say if you have a list and ask them what number element is this? Again going back to mathematical programming languages, that is why they start with element index 1 for the first element because the numbers match up (reason for using a zero index is we would add that value to the starting memory index so we would want to start from 0 at the beginning, since it is zero distance away).